src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -802,6 +804,15 @@ private async Task TryNotifyClientErrorAsync(IClientProxy client, string error,
         });
     }
 
+    private static void RestoreAntiforgeryToken(AsyncServiceScope scope)
+    {
+        // GetAntiforgeryToken makes sure the antiforgery token is restored from persitent component
+        // state and is available on the circuit whether or not is used by a component on the first
+        // render.
```

Comment:
I'm glad to see this logic only pulls the token from persistent component state and will never end up re-invoking the `IAntiforgery` service. I'm guessing we're relying on nothing calling `SetRequestContext` in the interactive scope. Is there an easy way to verify the `IAntiforgery` isn't reused in a test?

Reply:
Not really, we are the ones calling it in any case.